sdk/compiler/daml-extension/src/extension.ts

```diff
@@ -318,10 +332,28 @@ function getLanguageServerArgs(
   return serverArgs;
 }
 
-export function createLanguageClient(
+async function getSdkVersion(
+  damlPath: string,
+  projectPath: string | undefined,
+): Promise<string | undefined> {
+  // Ordered by priority
+  const selectionStrings = [
+    "selected by env var ",
+    "project SDK version from daml.yaml",
+    "default SDK version for new projects",
+  ];
+  const { stdout } = await exec(damlPath + " version", { cwd: projectPath });
+  const lines = stdout.split("\n");
+  for (const selection of selectionStrings) {
+    const line = lines.find((line: string) => line.includes(selection));
+    if (line) return line.trim().split(" ")[0];
```

Comment:
this feels a bit brittle, since we might inadvertently change the `daml version` output. perhaps we could add a flag to `daml version` that prints _only_ the version of daml that's active in cwd - no need to address it now, though.

Reply:
The problem with that is that it won't work with old versions. I need this check to essentially work in all of 2.9 at least.
I had considered a check where I just run `daml multi-ide --help` first, and see if that launches
but 2.8 has a worse version of multi-ide :/

I agree that such a flag would be _very_ helpful. We could even have a `json` flag, to make the output info easier to consume

Reply:
aaaah yes backwards compatibility fun :(

Reply:
Endless fun for the whole family!

Reply:
Note to self, leave comment on the daml version code saying to update this